.circleci/workflows.yml

```diff
@@ -0,0 +1,1094 @@
+            # For more on twine, see:
+            #   https://twine.readthedocs.io/en/latest/
+            twine upload --skip-existing dist/*
+
+workflows:
+  version: 2
+  build:
+    jobs: &build_jobs
+      - manual-trigger-required-for-fork
+      - build
+      - verify-requirements
+      - verify-format-yaml
+      - deploy-to-pypi:
+          name: Deploy bqetl to PyPI
+          requires:
+            - build
+          filters:
+            tags:
+              only: /[0-9]{4}.[0-9]{1,2}.[0-9]+/  # Calver: YYYY.M.MINOR
+            branches:
+              # Ignore all branches; this workflow should only run for tags.
+              ignore: /.*/
+
+  ### Validate SQL
+  validate-sql:
+    when:
+      or:
+        - << pipeline.parameters.validate-sql >>
+        - << pipeline.parameters.validate-routines >>
+    jobs:
+      - verify-format-sql
+      - deploy-changes-to-stage:
+          requires:
+            - generate-sql
+      - dry-run-sql:
+          requires:
+            - deploy-changes-to-stage
+      - test-sql:
+          requires:
+            - deploy-changes-to-stage
+      - validate-metadata:
+          requires:
+            - deploy-changes-to-stage
+      - validate-dags:
+          requires:
+            - generate-dags
+      - validate-views:
+          requires:
+            - deploy-changes-to-stage
+      - generate-sql
+      - main-generate-sql-and-dags:
+          filters:
+            branches:
+              ignore: main
+      - generate-diff:
+          requires:
+            - generate-dags
+            - main-generate-sql-and-dags
+          filters:
+            branches:
+              ignore: main
+      - post-diff:
+          requires:
+            - generate-diff
+          filters:
+            branches:
+              ignore: main
+      - generate-dags:
+          requires:
+            - generate-sql
+      - docs:
+          requires:
+            - generate-sql
+          filters:
+            branches:
+              only: main
+      - push-generated-sql:
+          requires:
+            - validate-dags
+          filters:
+            branches:
+              only:
+                - main
+      - sync-dags-repo:
+          name: 🔃 Synchronize bigquery-etl submodule
+          repo-to-sync: ${CIRCLE_PROJECT_REPONAME}
+          target-branch: generated-sql
+          requires:
+            - push-generated-sql
+          filters:
+            branches:
+              only:
+                - main
+      - reset-stage-env:
+          requires:
+            - push-generated-sql
+            - test-sql
+            - validate-views
+            - validate-metadata
+            - dry-run-sql
+      - test-routines:
+          requires:
+            - deploy-changes-to-stage
+
+  ### Validate bqetl tooling
+  validate-bqetl:
+    when:
+      or:
+        - << pipeline.parameters.validate-bqetl >>
+    jobs:
+      - test-bqetl
+      - integration
+
+  run-all:
```

Comment:
What's the reason for going with `run-all` rather than using "gcr deploy" parameter and a workflow that runs only those jobs?  In that case builds on main can just run with all the params set to true.  My initial impression is the `run-all` workflow might be confusing or error-prone because of the jobs duplicated in the workflows

Reply:
The deployment tasks depend on the `generate-sql` and `private-generate-sql` tasks. I guess we could add the deployment tasks to the `validate-sql` workflow, but then it might be confusing as well that it runs validation and does deploys when run on main.
The other option would be to run the `generate-sql` tasks twice, when validating SQL and before doing the deploys, but that just feels like unnecessarily increasing CircleCI cost and run time.

There doesn't seem to be a way to wait for another workflow in CircleCI, only for other jobs

Reply:
Ok I see the issue there.  Some jobs are already set with filters to run only on main so I don't think it's too much of a stretch to put the deploys in the other workflow and only run on main.  Just an opinion though and I'm not completely against this setup.  My current concern is that we always need to remember to add/remove jobs in both workflows and it wouldn't immediately be clear if we make a mistake and the workflows diverge.

Reply:
Yeah, I guess either way it's not ideal. It's possible that someone might add a task in a different workflow that deployment tasks depend on, which would not work then.
Maybe being more explicit and having different workflows is okay for now,

Reply:
Got it, I think this is good enough then.  Thanks